File: gzoom2-fe/app/src/app/api/service/party-role-view.service.ts
import { Injectable } from "@angular/core";
import { ApiClientService } from "app/commons/service/client.service";
import { PartyRoleView } from "../model/partyRoleView";
import { Observable, map } from "rxjs";

/**
 *  @author Leonardo Minaudo
 */
@Injectable({
  providedIn: 'root'
})
export class PartyRoleViewService {
  constructor(private client: ApiClientService) { }

  getPartyRoleView(
    statusId: string,
    organizationId: string
  ): Observable<PartyRoleView[]> {
    return this.client
      .get(`party-role-view/${statusId}/${organizationId}`)
      .pipe(map((json) => json.results as PartyRoleView[]));
  }

  getPartyRoleViewRoleTypeId(
    statusId: string,
    organizationId: string,
    roleTypeId: string
  ): Observable<PartyRoleView[]> {
    return this.client
      .get(`party-role-view/${statusId}/${organizationId}/${roleTypeId}`)
      .pipe(map((json) => json.results as PartyRoleView[]));
  }

  getPartyRoleViewByRoleTypeId(
    roleTypeId: string
  ): Observable<PartyRoleView[]> {
    return this.client
      .get(`party-role-view/${roleTypeId}`)
      .pipe(map((json) => json.results as PartyRoleView[]));
  }
}

File: gzoom2-be/rest/src/main/java/it/mapsgroup/gzoom/service/PartyRoleViewService.java
package it.mapsgroup.gzoom.service;

import it.mapsgroup.gzoom.model.Result;
import it.mapsgroup.gzoom.mybatis.dao.PartyRoleViewDao;
import it.mapsgroup.gzoom.mybatis.dto.PartyRoleView;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * @author Leonardo Minaudo
 */
@Service
public class PartyRoleViewService {
    private final PartyRoleViewDao partyRoleViewDao;

    @Autowired
    public PartyRoleViewService(PartyRoleViewDao partyRoleViewDao) {
        this.partyRoleViewDao = partyRoleViewDao;
    }

    public Result<PartyRoleView> getPartyRoleView(String statusId, String organizationId) {
        List<PartyRoleView> list = this.partyRoleViewDao.getPartyRoleView(statusId, organizationId);
        return new Result<>(list, list.size());
    }

    public Result<PartyRoleView> getPartyRoleViewRoleTypeId(String statusId, String organizationId, String roleTypeId) {
        List<PartyRoleView> list = this.partyRoleViewDao.getPartyRoleViewRoleTypeId(statusId, organizationId, roleTypeId);
        return new Result<>(list, list.size());
    }

    public Result<PartyRoleView> getPartyRoleViewByRoleTypeId(String roleTypeId) {
        List<PartyRoleView> list = this.partyRoleViewDao.getPartyRoleViewByRoleTypeId(roleTypeId);
        return new Result<>(list, list.size());
    }
}